Source/MQTTnet/Formatter/MqttBufferWriter.cs

```diff
@@ -63,17 +63,24 @@ public byte[] GetBuffer()
             return _buffer;
         }
 
-        public static int GetLengthOfVariableInteger(uint value)
+        public static int GetVariableByteIntegerSize(uint value)
         {
-            var result = 0;
-            var x = value;
-            do
+            if (value <= 127)
             {
-                x /= 128;
-                result++;
-            } while (x > 0);
+                return 1;
+            }
 
-            return result;
+            if (value <= 16383)
```

Comment:
Can this be expressed as `0x7F`, `0x3F_FF`, `0x1F_FF_FF`? To make it a bit more obvious where these constants come from? The old algorithm was clearer on that regard.
Or as comment just a link to the spec that dictates that encoding.

Reply:
Also when I read https://github.com/dotnet/MQTTnet/blob/00481533c13560f04330bcfb4804f1ba90174293/Source/MQTTnet.AspnetCore/ReaderExtensions.cs#L107 correct and as I would assume only the most significant bit is relevant. 
With the constants used it's 1, 2, 3 bits that are "reserved for continuation", while only 1 is needed?
So the constants should be `0x7F`, `0x7F_FF`, `0x7F_FF_FF`? So bigger ranges can be encoded with less bytes used. Or I'm missing something...

Reply:
I imported some documentation from the RFC to explain the values.

I do not really unterstand what you are referring to in the _ReaderExtensions_.

Reply:
I will merge this PR so that I can release a hotfix version soon. We can discuss this in another branch.

Reply:
> I do not really unterstand what you are referring to in the ReaderExtensions.

Sorry, I mixed the plain and the encoding threshoulds.